File: 백준/Silver/1978. 소수 찾기/소수 찾기.py
import math
import sys
input = sys.stdin.readline

n = int(input())
numList = list(map(int, input().split()))

d = [ True for i in range(1001)] # true 소수

d[0:1] = False, False

for i in range(2, int(math.sqrt(1000))+1) :
    if d[i] == True :
        for j in range(i+i, 1001, i) :
            if d[j] == True :
                d[j] = False

count = 0
for i in numList :
    if d[i] == True :
        count += 1

print(count)

File: 백준/Silver/1978. 소수 찾기/소수 찾기.java

import java.io.*;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		ArrayList<Integer> nums = new ArrayList<>();
		
		StringTokenizer stk = null;
		stk = new StringTokenizer(br.readLine(), " ");
		for(int a=0; a<n; a++) {
			nums.add(Integer.parseInt(stk.nextToken()));
		}
		
		int[] arr = new int[1001]; // 0 - 소수
		
		for(int i=2; i<Math.sqrt(1000)+1; i++) {
			if (arr[i] == 0) {
				for(int j=i+i; j<1001; j+=i) {
					// System.out.println(j);
					if (arr[j] == 0) {
						arr[j] = 1;
					}
				}
			}
		}
		arr[0] = 1;
		arr[1] = 1;
		
		int count = 0;
		for (Integer a : nums) {
			//System.out.println(a);
			if(arr[a] == 0) {
				count++;
			}
		}
		System.out.println(count);
		
	}
}
